### PR TITLE
[feature](nereides) support sort translator

### PR DESCRIPTION
**Pull request by zhengshiJ**
# Proposed changes

Physical sort:
     * 1. Build sortInfo
     *    There are two types of slotRef:
     *    one is generated by the previous node, collectively called old.
     *    the other is newly generated by the sort node, collectively called new.
     *    Filling of sortInfo related data structures,
     *    a. ordering use newSlotRef.
     *    b. sortTupleSlotExprs use oldSlotRef.
     * 2. Create sortNode
     * 3. Create mergeFragment

TODO:
1.Currently, columns that do not exist in select but exist in order by cannot be parsed.
eg: select key from table order by value;

2.For the combination of Literal and slotRefrance in select, there is a problem with parsing, 
eg: select key ,(10-value) from table;

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (No)
2. Has unit tests been added: (No)
3. Has document been added or modified: (No Need)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 1]
```diff
@@ -255,7 +279,24 @@ public PlanFragment visitPhysicalSort(PhysicalUnaryPlan<PhysicalHeapSort, Plan>
...
+     * Populate the SortTupleSlotExprs member.
+     */
+    private void createSortTupleSlotExprs(SortInfo info,
+            TupleDescriptor tupleDescriptor,
+            List<Expr> outputList, List<Expr> orderingExpr) {
+        List<Expr> sortTupleSlotExprs = new ArrayList<>();
+        Set<ExprId> exprIds = new HashSet<>();
+        outputList.addAll(orderingExpr);
+        for (Expr expr : outputList) {
+            if (!exprIds.contains(expr)) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:292]: exprIds never update. and use ExprId compare with Expr

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 2]
```diff
@@ -362,6 +403,19 @@ private TupleDescriptor generateTupleDesc(List<Slot> slotList, PlanTranslatorCon
         return tupleDescriptor;
     }
 
+    private TupleDescriptor generateTupleDesc(List<Slot> slotList, List<OrderKey> orderKeyList,
```
[the remark this answers is not in the document]
Comment (zhengshiJ) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:401]: No, the ones that have already been generated will be deduplicated, So only a,b,c will be generated

File: fe/fe-core/src/main/java/org/apache/doris/analysis/SortInfo.java [hunk 3]
```diff
@@ -63,6 +63,7 @@ public class SortInfo {
     // sortTupleDesc_.
     private List<Expr> sortTupleSlotExprs;
 
+
```
[the remark this answers is not in the document]
Comment (zhengshiJ) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/SortInfo.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java (hunk 1 above)
Comment (zhengshiJ) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/analysis/SortInfo.java [hunk 4]
```diff
@@ -117,6 +119,24 @@ public void setMaterializedTupleInfo(
         }
     }
 
+    /**
+     * Sets tupleInfo.
+     * Just for Nereids.
+     */
+    public void setTupleInfo(
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/SortInfo.java:126]: Wraps up as an implementation of the expr like ```finalizeImplForNereids``` function, and annotated only for the nereids optimizer translator.

Comment (zhengshiJ) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/SortInfo.java]: done

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**